openeo_driver/jobregistry.py

```diff
@@ -216,6 +229,9 @@ def __init__(
         session: Optional[requests.Session] = None,
         _debug_show_curl: bool = False,
     ):
+        if api_url is None:
```

Comment:
you could even make this 
```suggestion
        if not api_url:
```
to also fail on empty strings (or other "falsy" values)